src/variousZZZ.cc

```diff
@@ -0,0 +1,397 @@
+{
+  ARTS_USER_ERROR_IF(gfield3.get_grid_name(0) != "Altitude",
+                     "Name of first grid must be \"Altitude\".");
+  ARTS_USER_ERROR_IF(gfield3.get_grid_name(1) != "Latitude",
+                     "Name of second grid must be \"Latitude\".");
+  ARTS_USER_ERROR_IF(gfield3.get_grid_name(2) != "Longitude",
+                     "Name of third grid must be \"Longitude\".");
+  gfield3.checksize_strict();
+  chk_rte_pos("pos", pos);
+  
+  value = interp_gfield3(gfield3, pos);
+}                  
+
+
+/* Workspace method: Doxygen documentation will be auto-generated */
+void NumericInterpLatLonField(Numeric& value,
+                              const GriddedField2& gfield2,
+                              const Vector& pos,
+                              const Verbosity&)
+{
+  ARTS_USER_ERROR_IF(gfield2.get_grid_name(0) != "Latitude",
+                     "Name of first grid must be \"Latitude\".");
+  ARTS_USER_ERROR_IF(gfield2.get_grid_name(1) != "Longitude",
+                     "Name of second grid must be \"Longitude\".");
+  gfield2.checksize_strict();
+  chk_rte_pos("pos", pos);
+  
+  value = interp_gfield2(gfield2, Vector{pos[Range(1, 2)]});
+}                  
+
+
+//
+// So far a local solution 
+//
+void gridpos_local(ArrayOfGridPos& gp,
+                 ConstVectorView grid, 
+                 ConstVectorView points) {
+  const Index n = points.nelem();
+  ARTS_ASSERT(gp.nelem() == n);
+
+  // To save time in case of grid length 1
+  const Index l = grid.nelem();
+  if (l == 1) {
+    gp4length1grid(gp);
+    return;
+  }
+    
+  for (Index i=0; i<n; ++i) {
+    Numeric x = points[i];
+    
+    // Handle nearest extrapolation
+    if (x < grid[0])
+      x = grid[0];
+    else if (x > grid[l - 1])
+      x = grid[l - 1];
+
+    gridpos(gp[i], grid, x);
+  }
+}
+
+
+/* Workspace method: Doxygen documentation will be auto-generated */
+void NumericInterpVector(Numeric& value,
+                         const Vector& x,
+                         const Vector& y,
+                         const Numeric& xv,
+                         const Verbosity&)
+{
+  ArrayOfGridPos gp(1);
+  gridpos_local(gp, x, Vector(1, xv));
+  Vector itw(2);
+  interpweights(itw, gp[0]);
+  value = interp(itw, y, gp[0]);
+}
+
+
+Numeric interp_gfield2(const GriddedField2& G,
+                       const Vector& pos2D)
+{
+  ARTS_ASSERT(pos2D.nelem() == 2);
+  ARTS_ASSERT(G.checksize());
+
+  // Sizes
+  const Index ncols = G.data.ncols();
+  const Index nrows = G.data.nrows();
+
+  // To save time in case of data size (1, 1)
+  if (nrows == 1 && ncols == 1) {
+    return G.data(0, 0);
+  }
+  
+  // Get grid positions
+  ArrayOfGridPos gp_row(1), gp_col(1);
+  gridpos_local(gp_row, G.get_numeric_grid(0), Vector(1, pos2D[0]));
+  gridpos_local(gp_col, G.get_numeric_grid(1), Vector(1, pos2D[1]));
+
+  // Interpolate
+  Vector itw(4);
+  interpweights(itw, gp_row[0], gp_col[0]);
+  return interp(itw, G.data, gp_row[0], gp_col[0]);  
```

Comment:
@erikssonpatrick There is bug here that is uncovered in debug mode by new asserts in the latest matpack. G.data must have at least 2 elements in each dimension for the red interpolation to work. Same applies to `interp_gfield3`.

Reply:
Great that new matpack is better on finding bugs! This could be avoided with some local if-statements. But that would be an ugly fix. The real solution would be to update the interpolation code to handle this. And other types of possible extrapolations, we will introduce.
It is said that we will be able to set O2 VMR by a scalar value. I see this as a nearest extrapolation for length-1 grid.  And my temporary solution here tried to allow the same for GF2. As I have said, we should be as consistent with inter- and extra-polation as possible. 
Another issue here is that the second dimension often will be longitude, and then the circular interpolation started by Richard should be applied. How to flag when that interpolation should be done?
@riclarsson @olemke A meeting to discuss how to handle this? To handle atmospheric and surface data consistently.

Reply:
@erikssonpatrick Sure, we can have quick meeting.  I agree we should be consistent here.

The atmospheric fields keeps values like O2-VMR being constant as a simple `Numeric` so it is not applicable here, where you have fields to extrapolate from.  Perhaps you want a similar solution for the surface?

Reply:
@erikssonpatrick When does it fit for you Patrick? How about today at 14:00? We're also both available tomorrow if you prefer that.

Reply:
This is not very urgent from my side. But no free time Tue or Wed. Today 14.00 works, so maybe best to take it despite I will have no time to prepare. Otherwise any time Thu is fine.